lib/Dialect/SPIRV/SPIRVOps.cpp

```diff
@@ -733,6 +733,34 @@ static LogicalResult verify(spirv::BitcastOp bitcastOp) {
   return success();
 }
 
+namespace {
+
+// Convert chained `spirv::BitcastOp` operations into one
+// `spirv::BitcastOp` operation.
+struct ConvertChainedBitcast : public OpRewritePattern<spirv::BitcastOp> {
+  using OpRewritePattern<spirv::BitcastOp>::OpRewritePattern;
+
+  PatternMatchResult matchAndRewrite(spirv::BitcastOp bitcastOp,
+                                     PatternRewriter &rewriter) const override {
+    auto parentBitcastOp = dyn_cast_or_null<spirv::BitcastOp>(
+        bitcastOp.operand()->getDefiningOp());
+
+    if (!parentBitcastOp) {
+      return matchFailure();
+    }
+
+    rewriter.replaceOpWithNewOp<spirv::BitcastOp>(
```

Comment:
You could use the version of replaceOpWithNewOp that takes "values to remove if dead" as its first arg to potentially also get rid of parentBitcastOp. This will make it converge faster.